### PR TITLE
Fix Mistral chat completion

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

We've observed several failures related to querying Mistral models, which appear to be tied to how we construct the chat completion object. Specifically, we've been including a `name` attribute for all completions, except that Mistral's API does not support this parameter ([doc](https://docs.mistral.ai/api/#operation/createChatCompletion)).

After some testing, `name` is only accepted when the role is set to function/tool. This PR adjusts our approach accordingly. To note: despite this change, we continue to inject the user/assistant name into the completion through the meta prompt.

### CODE REVIEW

File: core/src/providers/mistral.rs [hunk 1]
```diff
@@ -154,7 +154,11 @@ impl TryFrom<&ChatMessage> for MistralChatMessage {
                 meta_prompt,
                 cm.content.clone().unwrap_or(String::from(""))
             )),
-            name: cm.name.clone(),
+            // Name is only supported for the Function/Tool role.
+            name: match cm.role {
+                ChatMessageRole::Function => cm.name.clone(),
+                _ => None,
```
Comment (fontanierh) [inline core/src/providers/mistral.rs:160]: This will end up as `role: null` in the json no ?

Comment (flvndvd) [reply core/src/providers/mistral.rs:160]: Nope from my test, the key is removed.

Comment (flvndvd) [reply core/src/providers/mistral.rs:160]: It's handled by:
https://github.com/dust-tt/dust/blob/c1af4a7317302fe1773854dde30e6c0185f9fc2f/core/src/providers/mistral.rs#L99-L100

Comment (fontanierh) [reply core/src/providers/mistral.rs:160]: We may want to omit the key entirely instead

Comment (flvndvd) [reply core/src/providers/mistral.rs:160]: I mean it's the whole point of using serde, right 🙃 ?

Comment (fontanierh) [reply core/src/providers/mistral.rs:160]: Indeed, didn't see that (and didn't know serde had  that)

**Review by fontanierh — APPROVED** [review]

File: core/src/providers/mistral.rs (hunk 1 above)
Comment (fontanierh) [reply core/src/providers/mistral.rs:160]: (hadn't seen your reply either, github page wasn't refreshed)

**Pull request merged**